crates/router/src/core/payouts/helpers.rs

```diff
@@ -904,12 +906,20 @@ pub fn is_payout_terminal_state(status: api_enums::PayoutStatus) -> bool {
     !matches!(
         status,
         api_enums::PayoutStatus::Pending
+            | api_enums::PayoutStatus::Initiated
             | api_enums::PayoutStatus::RequiresCreation
             | api_enums::PayoutStatus::RequiresFulfillment
             | api_enums::PayoutStatus::RequiresPayoutMethodData
     )
 }
 
+pub fn should_call_retrieve(status: api_enums::PayoutStatus) -> bool {
+    matches!(
+        status,
+        api_enums::PayoutStatus::Pending | api_enums::PayoutStatus::Initiated
```

Comment:
This should have all the non-terminal states right?

Reply:
these are the only two non-terminal status where we can retrieve, in case of `require_creation`/`require_confirmation`, since we don't hit the connector, we can't retrieve it. in case of `require_fulfillment`, we don't hit the connector for some of the connector ( creating payout object only at our end) thus not able to retrieve in those cases